lib/internal/http2/core.js

```diff
@@ -1416,86 +1546,141 @@ class Http2Stream extends Duplex {
     validatePriorityOptions(options);
 
     const priorityFn = submitPriority.bind(this, options);
-    if (this[kID] === undefined) {
+
+    // If the handle has not yet been assigned, queue up the priority
+    // frame to be sent as soon as the ready event is emitted.
+    if (this.pending) {
       this.once('ready', priorityFn);
       return;
     }
     priorityFn();
   }
 
+  get closed() {
+    return !!(this[kState].flags & STREAM_FLAGS_CLOSED);
+  }
+
+  // Close initiates closing the Http2Stream instance by sending an RST_STREAM
+  // frame to the connected peer. The readable and writable sides of the
+  // Http2Stream duplex are closed and the timeout timer is unenrolled. If
+  // a callback is passed, it is registered to listen for the 'close' event.
+  //
+  // If the handle and stream ID have not been assigned yet, the close
+  // will be queued up to wait for the ready event. As soon as the stream ID
+  // is determined, the close will proceed.
+  //
+  // Submitting the RST_STREAM frame to the underlying handle will cause
+  // the Http2Stream to be closed and ultimately destroyed. After calling
+  // close, it is still possible to queue up PRIORITY and RST_STREAM frames,
+  // but no DATA and HEADERS frames may be sent.
+  close(code = NGHTTP2_NO_ERROR, callback) {
```

Comment:
Can this be  `close(callback)`? I would prefer to not expose NGHTTP2 error codes. For errors, we have `destroy()`.

Reply:
This is not an NGHTTP2 error code, it's an HTTP2 RST_STREAM error code. An RST_STREAM is used to close the stream here so we need to know what code to use. There are legitimate cases where the user might specify something other than `NO_ERROR` without using `destroy(err)`